pkg/config/redisstore.go

```diff
@@ -163,4 +256,89 @@ func (rs *redisStore) Close() {
 	if err := rs.client.Close(); err != nil {
 		glog.Warningf("failed to close the connection: %v", err)
 	}
+	if rs.redisSubscriber != nil {
+		rs.redisSubscriber.Close()
+	}
+}
+
+func (rs *redisStore) IsStoreChangeAvailable() bool {
+	return rs.redisSubscriber != nil
+}
+
+func (sub *redisSubscriber) startListening(dbNum uint64) error {
+	pattern := fmt.Sprintf("__keyevent@%d__:*", dbNum)
+	resp := sub.client.PSubscribe(pattern)
+	if resp.Err != nil {
+		return resp.Err
+	}
+	sub.running = true
+	go func() {
+		for {
+			resp = sub.client.Receive()
+			if !sub.running {
+				break
+			}
+			if resp.Type == pubsub.Error {
+				if resp.Timeout() {
+					continue
+				}
+				glog.Warningf("unexpected error on subscription: %v", resp.Err)
+				break
+			}
+			if resp.Type == pubsub.Message {
+				op := resp.Channel[strings.Index(resp.Channel, ":")+1:]
+				c := Change{
+					Index: sub.lastIndex + 1,
+					Key:   resp.Message,
+				}
+				if op == "set" {
+					c.Type = Update
+				} else if op == "del" {
+					c.Type = Delete
+				} else {
+					continue
+				}
+				sub.mu.Lock()
+				sub.changes = append(sub.changes, c)
+				sub.lastIndex++
+				l := sub.listener
+				sub.mu.Unlock()
+				if l != nil {
+					l.NotifyStoreChanged(sub.lastIndex)
+				}
+			}
+		}
+		sub.client.PUnsubscribe(pattern)
+		_ = sub.client.Client.Close()
+		sub.client = nil
+	}()
+	return nil
+}
+
+func (sub *redisSubscriber) Close() {
+	sub.mu.Lock()
+	defer sub.mu.Unlock()
+	if sub.client == nil || !sub.running {
+		return
+	}
+	sub.running = false
+}
+
+func (sub *redisSubscriber) RegisterStoreChangeListener(s StoreListener) {
```

Comment:
RegisterStoreChangeListener -> RegisterListener

No need to mention StoreListener, this func is already unambiguous.